### PR TITLE
docs: manifests for SSO using ArgoCD Dex, to be used with Kustomize

### PR DESCRIPTION
**Pull request by bgdnlp**
Add some clarification around how secrets should be set and Kubernetes manifests as examples for those who don't deploy with Helm.

Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md).

### CODE REVIEW

File: docs/argo-server-sso.md [hunk 1]
```diff
@@ -112,13 +112,24 @@ If no rule matches, we deny the user access.
...
+    client-id: YXJnby13b3JrZmxvd3Mtc3Nv
+    # client-secret is 'MY-SECRET-STRING-CAN-BE-UUID'
+    client-secret: TVktU0VDUkVULVNUUklORy1DQU4tQkUtVVVJRA==
+  ```
 - `--auth-mode=sso` server argument added
 - A Dex `staticClients` configured for `argo-workflows-sso`
 - The `sso` configuration filled out in Argo Workflows Server to match
 
-What this might look like in your chart configuration:
+### Example Helm chart configuration for authenticating against ArgoCD's Dex
```
Comment (alexec) [inline docs/argo-server-sso.md:132]: can we put this in its own document please?

**Review by alexec — CHANGES_REQUESTED** [review]

File: docs/argo-server-sso.md (hunk 1 above)
Comment (bgdnlp) [reply docs/argo-server-sso.md:132]: What do you mean, which part? One document for Helm, another for the manifests? Or the entire SSO-via-ArgoCD-Dex section in its own document?

Comment (alexec) [reply docs/argo-server-sso.md:132]: one document for using Argo DEX

Comment (bgdnlp) [reply docs/argo-server-sso.md]: How does it look now?

**Review by alexec — APPROVED** [review]

**Comment by alexec** [comment]
Can you please signoff your commits to pass the DCO check?

**Comment by bgdnlp** [comment]
> Can you please signoff your commits to pass the DCO check?

I think I screwed that up, I merged from main repo before and now it signed off everything. I'm not sure what that means, I can close this pull request and redo the whole thing later, make it clean.

**Comment by alexec** [comment]
I think you need to resolve conflicts too?

**Review by alexec — CHANGES_REQUESTED** [review]
please fix conflicts

**Pull request closed without merging**